src/celeritas/optical/CerenkovDistribution.hh

```diff
@@ -0,0 +1,50 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/CerenkovDistribution.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Pre- and post-step data for sampling Cerenkov photons.
+ */
+struct CerenkovStepData
+{
+    real_type mean_num_photons;  //!< Average photons produced per unit length
```

Comment:
Good point, since this is not used to sample the number of photons there's no reason we can't calculate it in the generator. I'll add a helper implementing `G4Cerenkov::GetAverageNumberOfPhotons()`, which will also be used by the "pre-generator".